R/get_attribution.R

```diff
@@ -71,11 +81,58 @@ get_attribution <- function(uuid = NULL, img = NULL, text = FALSE) {
             "CC BY-NC-SA 3.0",
             "CC BY-NC 3.0")
   licenses <- data.frame(links, abbr)
-
+  # Create permalink ------------------------------------------------------
+  if (permalink) {
+    coll <- phy_POST(path = "collections", body = uuid)$uuid
+    url <- paste0("https://www.phylopic.org/api/permalinks/collections/", 
+                  coll)
+    coll <- GET(url = url) 
+    hash <- response_to_JSON(coll)
+    perm <- paste0("https://www.phylopic.org/permalinks/", hash)
+  }
   # API call -------------------------------------------------------------
   if (length(uuid) > 1) {
     att <- lapply(uuid, get_attribution)
     names(att) <- uuid
+    # Text output?
+    if (text) {
+      txt <- lapply(att, function(x) {
+        paste0(x$contributor, ", ",
+               substr(x$created, start = 1, stop = 4), " ",
+               "(", x$license_abbr, ")")
+      })
+      # Keep unique items
+      txt <- unique(unlist(txt))
+      # Convert to string
+      if (length(txt) > 1) {
+        txt <- combine_words(txt, oxford_comma = TRUE)
+        txt <- paste0("Silhouettes were contributed by ", toString(txt), ".")
+      } else {
+        txt <- paste0("Silhouette was contributed by ", toString(txt), ".")
+      }
+      txt <- paste0("Organism silhouettes are from PhyloPic ",
+                    "(https://www.phylopic.org/; T. Michael Keesey, 2023) ",
+                    "and were added using the rphylopic R package ver. ",
+                    packageVersion("rphylopic"), " (Gearty & Jones, 2023). ",
+                    txt)
+      # Add permalink?
+      if (permalink) {
+        txt <- paste0(txt, " Permalink: ", perm, ".")
+      }
+    }
+    # Additional lists needed?
+    if (text || permalink) {
+      att <- list(images = att)
```

Comment:
Hmm...not sure how I feel about the return behavior changing here based on these arguments. I suppose it is fine as long as this format is clear in the @return documentation, but I think it might be easiest to always have the images under an "images" list element.